main/src/ca/uwaterloo/flix/language/phase/LoadBytecode.scala

```diff
@@ -233,4 +320,67 @@ object LoadBytecode extends Phase[ExecutableAst.Root, ExecutableAst.Root] {
+    * Find enums from the given expression
+    * @param e expression
+    * @return
+    */
+  def findEnums(e: Expression): List[(Type, (String, Type))] = e match {
+    case Expression.Unit => Nil
+    case Expression.True => Nil
+    case Expression.False => Nil
+    case Expression.Char(lit) => Nil
+    case Expression.Float32(lit) => Nil
+    case Expression.Float64(lit) => Nil
+    case Expression.Int8(lit) => Nil
+    case Expression.Int16(lit) => Nil
+    case Expression.Int32(lit) => Nil
+    case Expression.Int64(lit) => Nil
+    case Expression.BigInt(lit) => Nil
+    case Expression.Str(lit) => Nil
+    case Expression.LoadBool(n, o) => Nil
+    case Expression.LoadInt8(b, o) => Nil
+    case Expression.LoadInt16(b, o) => Nil
+    case Expression.LoadInt32(b, o) => Nil
+    case Expression.StoreBool(b, o, v) => Nil
+    case Expression.StoreInt8(b, o, v) => Nil
+    case Expression.StoreInt16(b, o, v) => Nil
+    case Expression.StoreInt32(b, o, v) => Nil
+    case Expression.Var(sym, tpe, loc) => Nil
+    case Expression.Ref(name, tpe, loc) => Nil
+    case Expression.MkClosureRef(ref, freeVars, tpe, loc) => Nil
+    case Expression.ApplyRef(name, args, tpe, loc) => args.flatMap(findEnums)
+    case Expression.ApplyTail(name, formals, actuals, tpe, loc) => actuals.flatMap(findEnums)
+    case Expression.ApplyHook(hook, args, tpe, loc) => args.flatMap(findEnums)
+    case Expression.ApplyClosure(exp, args, tpe, loc) => findEnums(exp) ++ args.flatMap(findEnums)
+    case Expression.Unary(op, exp, tpe, loc) => findEnums(exp)
+    case Expression.Binary(op, exp1, exp2, tpe, loc) => findEnums(exp1) ++ findEnums(exp2)
+    case Expression.IfThenElse(exp1, exp2, exp3, tpe, loc) => findEnums(exp1) ++ findEnums(exp2) ++ findEnums(exp3)
+    case Expression.Let(sym, exp1, exp2, tpe, loc) => findEnums(exp1) ++ findEnums(exp2)
+    case Expression.Is(sym, tag, exp, loc) => Nil
+    case Expression.Tag(enum, tag, exp, tpe, loc) => List((tpe, (tag, exp.tpe))) ++ findEnums(exp)
+    case Expression.Untag(sym, tag, exp, tpe, loc)  => Nil
+    case Expression.Index(base, offset, tpe, loc) => findEnums(base)
+    case Expression.Tuple(elms, tpe, loc) => elms.flatMap(findEnums).toList
+    case Expression.Existential(params, exp, loc) => findEnums(exp)
+    case Expression.Universal(params, exp, loc) => findEnums(exp)
+    case Expression.NativeConstructor(constructor, args, tpe, loc) => args.flatMap(findEnums)
+    case Expression.NativeField(field, tpe, loc) => Nil
+    case Expression.NativeMethod(method, args, tpe, loc) => args.flatMap(findEnums)
+    case Expression.UserError(tpe, loc) => Nil
+    case Expression.MatchError(tpe, loc) => Nil
+    case Expression.SwitchError(tpe, loc) => Nil
+  }
+
+  /** TODO: CHANGE THIS!
+    * Type to a string wrapped in a list. This string doesn't contain prohibited characters of the type representation
+    * @param tpe type
+    * @return
+    */
+  private def typeToUniquePath(tpe: Type) : List[String] = {
```

Comment:
We will create `ca.waterloo.flix.enums.Int32.object.some`.
The general formatting is `ca.waterloo.flix.enums.NAMESPACE.ENUMNAME.TYPEOFFIELD.CASENAME`. `ca.waterloo.flix.enums` is to differentiate generated codes from other codes, `NAMESPACE.ENUMNAME` should uniquely identify the enum. `TYPEOFFIELD` should uniquely identify type of the value field and `CASENAME` must be unique too.